arrays/arrays.c

```diff
@@ -47,9 +53,11 @@ void destroy_array(Array *arr) {
 void resize_array(Array *arr) {
 
   // Create a new element storage with double capacity
-
+  Array new_arr = create_array((arr->capacity * 2));
```

Comment:
This is interesting, you might also consider creating a new char** and replacing array->elements with that. If you are going to do this though remember to destroy the old array after you create a new one.

Reply:
I eventually came around to that on this one